Modules/QuestieInit.lua

```diff
@@ -226,20 +255,37 @@ function _QuestieInit:OverrideDBWithTBCData()
     end
 end
 
--- called by the PLAYER_LOGIN event handler
--- this function creates the coroutine that runs "InitAllModules"
-function QuestieInit:Init()
+
+-- this function creates coroutine to run a function from QuestieInit.Stages[]
+---@param stage number @the stage to start
+function _QuestieInit:StartStageCoroutine(stage)
     local initFrame = CreateFrame("Frame")
-    local routine = coroutine.create(QuestieInit.InitAllModules)
-    initFrame:SetScript("OnUpdate", function()
-        local success, error = coroutine.resume(routine)
+    local routine = coroutine.create(QuestieInit.Stages[stage])
+
+    local function InitOnUpdate()
+        local success, ret = coroutine.resume(routine)
         if success then
             if coroutine.status(routine) == "dead" then
                 initFrame:SetScript("OnUpdate", nil)
+                initFrame:SetParent(nil)
+                initFrame = nil
+                if type(ret) == "function" then -- continue to next stage, if it was returned by coroutine
+                    ret()
+                end
```

Comment:
This is optional, because some stages want to return a function which can/has to run right after the previous stage completed?
Can't the init stages just be looped or all functions be started as coroutine? 🤔

Reply:
Some stages might not want to be coroutines. For example while cache checking was still its own stage and not done async and earlier, it was not a coroutine

Reply:
Another option would assume just iterating all stages in order and add some other table to tell about each if it is a coroutine or not.
Current one allows to move execution to where ever, not necessary into next stage. I didn't know what would be best option to do